packages/spectacular/src/lib/feature-testing/feature-harness/create-feature-harness.spec.ts

```diff
@@ -0,0 +1,158 @@
+import { Location } from '@angular/common';
+import { SpyLocation } from '@angular/common/testing';
+import { Component, Injectable, NgModule } from '@angular/core';
+import { ExtraOptions, ROUTER_CONFIGURATION, RouterModule } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+import { SpectacularAppComponent } from '../../application-testing/app-component/spectacular-app.component';
+import { featurePathToken } from '../configuration/feature-path.token';
+import { createFeatureHarness } from './create-feature-harness';
+
+@Component({
+  template: '',
+})
+class HeroesJobBoardComponent {}
+
+@Component({
+  template: '',
+})
+class HeroesJobListingComponent {}
+
+const jobListingPath = 'listing';
+
+@NgModule({
+  declarations: [HeroesJobBoardComponent, HeroesJobListingComponent],
+  imports: [
+    RouterModule.forChild([
+      {
+        path: '',
+        pathMatch: 'full',
+        component: HeroesJobBoardComponent,
+      },
+      {
+        path: jobListingPath,
+        component: HeroesJobListingComponent,
+      },
+    ]),
+  ],
+})
+class HeroesJobBoardModule {}
+
+const featurePath = 'heroes-job-board';
+
+describe(createFeatureHarness.name, () => {
+  describe('Configuration', () => {
+    @Injectable()
+    class JobService {}
+
+    @NgModule({
+      providers: [JobService],
+    })
+    class JobServiceModule {}
+
+    it('adds the specified imports', () => {
+      const harness = createFeatureHarness({
+        featurePath,
+        featureModule: HeroesJobBoardModule,
+        imports: [JobServiceModule],
+      });
+
+      const jobService = harness.inject(JobService);
+      expect(jobService).toBeInstanceOf(JobService);
+    });
+
+    it('adds the specified providers', () => {
+      const harness = createFeatureHarness({
+        featurePath,
+        featureModule: HeroesJobBoardModule,
+        providers: [JobService],
+      });
+
+      const jobService = harness.inject(JobService);
+      expect(jobService).toBeInstanceOf(JobService);
+    });
+  });
+
+  describe('Routing', () => {
+    it(`imports the ${RouterTestingModule.name}`, () => {
+      const harness = createFeatureHarness({
+        featurePath,
+        featureModule: HeroesJobBoardModule,
+      });
+
+      const location = harness.inject(Location);
+      expect(location).toBeInstanceOf(SpyLocation);
```

Comment:
Isn't this part of the harness? `harness.location` ??

Reply:
You're probably right. I reused the tests from the previous implementation.

Reply:
@NachoVazquez, there's `Location` from `@angular/common` and there's `SpectacularFeatureLocation` from `@ngworker/spectacular`. Spectacular's feature location service wraps Angular's location service. However, the point of this test case is to illustrate that `RouterTestingModule` is being registered so that Angular's `Location` service is being replaced by `SpyLocation` from `@angular/common/testing`. For more information about this, see [Testing Angular routing components with the RouterTestingModule](https://dev.to/this-is-angular/testing-angular-routing-components-with-the-routertestingmodule-4cj0#angulars-routertestingmodule).